gdal.cmake

```
@@ -256,6 +256,14 @@ function (gdal_add_private_link_libraries)
 endfunction (gdal_add_private_link_libraries)
 
 add_library(${GDAL_LIB_TARGET_NAME} gcore/gdal.h)
+
+# If a shared lib renaming has been set in ConfigUser.cmake
+if (GDAL_SHAREDLIB_OUTPUT_NAME)
+  set_target_properties(GDAL PROPERTIES RUNTIME_OUTPUT_NAME ${GDAL_SHAREDLIB_OUTPUT_NAME})
```

```suggestion
  set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES RUNTIME_OUTPUT_NAME ${GDAL_SHAREDLIB_OUTPUT_NAME})
```

But actually the if() else () thing could be avoided if you do
```
set(GDAL_LIB_OUTPUT_NAME "gdal" CACHE STRING "Name of the GDAL library")
set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES OUTPUT_NAME ${GDAL_LIB_OUTPUT_NAME})
```
GDAL_LIB_OUTPUT_NAME is a more general name, since it can also apply to static builds

---

Had to revert this because it I did the suggestion wrong or it doesn't work.